pxf/pxf-api/src/main/java/org/apache/hawq/pxf/api/FilterParser.java

```diff
@@ -418,6 +430,40 @@ private Object parseParameter() throws Exception {
         return data;
     }
 
+    private Object parseListParameter() throws Exception {
+        if (index == filterByteArr.length) {
+            throw new FilterStringSyntaxException("argument should follow at " + index);
+        }
+
+        DataType dataType = DataType.get(parseConstDataType());
+        List<Object> data = new ArrayList<Object>();
+        if (dataType == DataType.UNSUPPORTED_TYPE) {
+            throw new FilterStringSyntaxException("invalid DataType OID at " + (index - 1));
+        }
+
+        if (dataType.getTypeElem() == null) {
+            throw new FilterStringSyntaxException("expected non-scalar datatype, but got datatype with oid = " + dataType.getOID());
+        }
+
+        while (((char) filterByteArr[index]) == CONST_LEN) {
```

Comment:
Also make sure we don't exeed filterByteArr.length in case the filterString is incorrect we can catch it